windows/src/global/delphi/general/VKeys.pas

```diff
@@ -294,6 +294,269 @@ function FindVKeyName(const txt: string): Word;
 	'K_?FE',				// &HFE
 	'K_?FF');				// &HFF
 
+///
+/// Virtual key names used in KeymanWeb. Blanks indicate a
+/// virtual key that is not recognised by KeymanWeb.
+///
+const KMWVKeyNames: array[0..255] of string = (
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  'K_BKSP',
+  'K_TAB',
+  '',
+  '',
+  '',
+  'K_ENTER',
+  '',
+  '',
+  'K_SHIFT',
+  'K_CONTROL',
+  'K_ALT',
+  'K_PAUSE',
+  'K_CAPS',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  'K_ESC',
+  '',
+  '',
+  '',
+  '',
+  'K_SPACE',
+  'K_PGUP',
+  'K_PGDN',
+  'K_END',
+  'K_HOME',
+  'K_LEFT',
+  'K_UP',
+  'K_RIGHT',
+  'K_DOWN',
+  'K_SEL',
+  'K_PRINT',
+  'K_EXEC',
+  '',
+  'K_INS',
+  'K_DEL',
+  'K_HELP',
+  'K_0',
+  'K_1',
+  'K_2',
+  'K_3',
+  'K_4',
+  'K_5',
+  'K_6',
+  'K_7',
+  'K_8',
+  'K_9',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  'K_A',
+  'K_B',
+  'K_C',
+  'K_D',
+  'K_E',
+  'K_F',
+  'K_G',
+  'K_H',
+  'K_I',
+  'K_J',
+  'K_K',
+  'K_L',
+  'K_M',
+  'K_N',
+  'K_O',
+  'K_P',
+  'K_Q',
+  'K_R',
+  'K_S',
+  'K_T',
+  'K_U',
+  'K_V',
+  'K_W',
+  'K_X',
+  'K_Y',
+  'K_Z',
+  '',
+  '',
+  '',
+  '',
+  '',
+  'K_NP0',
+  'K_NP1',
+  'K_NP2',
+  'K_NP3',
+  'K_NP4',
+  'K_NP5',
+  'K_NP6',
+  'K_NP7',
+  'K_NP8',
+  'K_NP9',
+  'K_NPSTAR',
+  'K_NPPLUS',
+  'K_SEPARATOR',
+  'K_NPMINUS',
+  'K_NPDOT',
+  'K_NPSLASH',
+  'K_F1',
+  'K_F2',
+  'K_F3',
+  'K_F4',
+  'K_F5',
+  'K_F6',
+  'K_F7',
+  'K_F8',
+  'K_F9',
+  'K_F10',
+  'K_F11',
+  'K_F12',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  'K_NUMLOCK',
+  'K_SCROLL',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  'K_LSHIFT',
+  'K_RSHIFT',
+  'K_LCONTROL',
+  'K_RCONTROL',
+  'K_LALT',
+  'K_RALT',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  'K_COLON',
+  'K_EQUAL',
+  'K_COMMA',
+  'K_HYPHEN',
+  'K_PERIOD',
+  'K_SLASH',
+  'K_BKQUOTE',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  'K_LBRKT',
+  'K_BKSLASH',
+  'K_RBRKT',
+  'K_QUOTE',
+  '',
+  '',
+  '',
+  'K_oE2',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  '',
+  ''
+);
```

Comment:
How are these osk.keyCodes handled?

```
      "K_LOPT":50001,"K_ROPT":50002,
      "K_NUMERALS":50003,"K_SYMBOLS":50004,"K_CURRENCIES":50005,
      "K_UPPER":50006,"K_LOWER":50007,"K_ALPHA":50008,
      "K_SHIFTED":50009,"K_ALTGR":50010,
      "K_TABBACK":50011,"K_TABFWD":50012
```

Reply:
Those are special-case key codes that are never used in key rules. So they don't need to be in the allowable key map. They are used only in touch keyboard layout descriptions.